src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -205,7 +205,8 @@ protected function addRoute(RouteCollection $collection, object $annot, array $g
             }
             foreach ($paths as $locale => $path) {
                 if (preg_match(sprintf('/\{%s(?:<.*?>)?\}/', preg_quote($param->name)), $path)) {
-                    $defaults[$param->name] = $param->getDefaultValue();
+                    $defaultValue = $param->getDefaultValue();
+                    $defaults[$param->name] = $defaultValue instanceof \BackedEnum ? $defaultValue->value : $defaultValue;
```

Comment:
We have rejected this kind of changes in the past. Please call `->value` on your enum case yourself if you need the backing value.

Reply:
I can't do that in my own code, as stated in #49203, if I set a default value in my function is has to be an Enum, but the router asks for a string.

Reply:
To be honest, to me it's more of a bug rather than a feature: as long as Enum is supported for route generation, the behavior that crashes in my issue (setting a default Enum in function parameters) is a bug. I understand that it's easier to port as a new feature but really this should not be a userland problem. 

A lot of it still stems from https://github.com/php/php-src/pull/8825 where Enums don't get their `__toString` method.

Reply:
@derrabus this code path is a place where the loader *guesses* the default based on the controller default value.

In the associated issue, I explicitly asked to change only that place and njot adding support for using enums directly in the Route attribute.

Reply:
The only other solution would be to make that guesser skip any default value that is not a scalar (to avoid generating an invalid default). But this would make DX worse.

Reply:
> this code path is a place where the loader guesses the default based on the controller default value.

Got it, makes sense to me, this plays well with the `BackedEnumValueResolver`!

:+1: on my side once tests are added.

Reply:
> make that guesser skip any default value that is not a scalar

shouldn't we do this?
```php
$defaults[$param->name] = match (true) {
    $defaultValue instanceof \BackedEnum => $defaultValue->value,
    $defaultValue instanceof \Stringable => (string) $defaultValue,
    \is_object($defaultValue) => null,
    default => $defaultValue,
};
```

Reply:
I don't think Stringable objects should be casted to create a default. The controller would then receive the string value from the request attributes and not the object that it has a default value.

And having `null` as default value would be a bad idea as well, as that might not be acceptable by the controller (a `null` default in the route is not the same than no default)

Reply:
We should we then just skip objects (and non scalars basically)?
```php
if (\is_scalar($defaultValue = $param->getDefaultValue()) {
    $defaults[$param->name] = $defaultValue;
} elseif ($defaultValue instanceof \BackedEnum) {
    $defaults[$param->name] = $defaultValue->value;
}
```

Reply:
That's probably better than guessing an invalid default, especially as we don't fail during loading for such case but only during route generation later.

Reply:
I'll update the code and make some test cases :+1: